File: 算法学习/数组、链表、跳表/删除排序数组中的重复项.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Time : 2020/4/15 20:36
# @Author : ZhouChuang
# @Site : 
# @File : 删除排序数组中的重复项.py
# @Software: PyCharm
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 思路1. 快慢指针，i是慢指针，j为快指针。如果nums[i]=nums[j],增加j以跳过。如果nums[j]!=nums[i],将快指针处的值复制到i+1处，
        #       最后返回的数为i+1
        if len(nums) == 0:
            return 0
        i = 0
        for j in range(1, len(nums)):
            if nums[j] != nums[i]:
                i += 1
                nums[i] = nums[j]
        return i + 1


if __name__ == '__main__':
    Sol = Solution()
    ans = Sol.removeDuplicates([1, 1, 2])
    print(ans)

File: 算法学习/数组、链表、跳表/删除排序数组中的重复项.go
/**
* @Author: ZhouChuang
* @Date: 2020/4/15 21:09
 */
package main

func removeDuplicates(nums []int) int {
	i := 0
	for j := 1; j < len(nums); j++ {
		if nums[j] != nums[i] {
			i++
			nums[i] = nums[j]
		}
	}
	return i + 1
}

func main() {
	nums := []int{1, 1, 3}
	removeDuplicates(nums)

}
